apps/web/components/eventtype/EventSetupTab.tsx

```diff
@@ -256,21 +256,22 @@ export const EventSetupTab = (
 
     const [showEmptyLocationSelect, setShowEmptyLocationSelect] = useState(false);
     const [selectedNewOption, setSelectedNewOption] = useState<SingleValueLocationOption | null>(null);
-
     return (
       <div className="w-full">
         <ul ref={animationRef} className="space-y-2">
           {locationFields.map((field, index) => {
             const eventLocationType = getEventLocationType(field.type);
-            const defaultLocation = formMethods
-              .getValues("locations")
-              ?.find((location: { type: EventLocationType["type"]; address?: string }) => {
-                if (location.type === LocationType.InPerson) {
-                  return location.type === eventLocationType?.type && location.address === field?.address;
-                } else {
-                  return location.type === eventLocationType?.type;
-                }
-              });
+            const defaultLocation = formMethods.getValues("locations")?.find((location) => {
+              if (eventLocationType?.organizerInputType) {
+                return (
+                  location.type === eventLocationType?.type &&
+                  location[eventLocationType.defaultValueVariable] ===
+                    field?.[eventLocationType.defaultValueVariable]
+                );
+              } else {
+                return location.type === eventLocationType?.type;
+              }
+            });
```

Comment:
1. This seems like abstractable to a function. 
2. I am having a hard time understanding why we are using both `locationFields` and `formMethod.getValues('locations')`, they seem to be having the same value.
3. What is the case when the two would be different?
4. What is the case when `location.type === eventLocationType?.type` will be false.